src/components/Equity/EquityStakingRow.tsx

```diff
@@ -61,6 +64,52 @@ export const EquityStakingRow: React.FC<EquityStakingRowProps> = ({
 
   const asset = useAppSelector(state => selectAssetById(state, assetId))
 
+  const underlyingAssetsWithBalancesAndIcons = useAppSelector(state =>
+    opportunity?.userStakingId
+      ? selectUnderlyingStakingAssetsWithBalancesAndIcons(state, {
+          userStakingId: opportunity.userStakingId,
+        })
+      : undefined,
+  )
+
+  const assetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, assetId),
+  )
+
+  const { cryptoAmountPrecision, fiatAmount } = useMemo(() => {
+    if (!opportunity || !asset || !assetMarketData)
+      return {
+        cryptoAmountPrecision: '0',
+        fiatAmount: '0',
+      }
+
+    const underlyingAssetBalance = underlyingAssetsWithBalancesAndIcons?.find(
+      ({ assetId: balanceAssetId }) => assetId === balanceAssetId,
+    )
+
+    if (!underlyingAssetBalance && underlyingAsset)
+      return {
+        cryptoAmountPrecision: bnOrZero(opportunity.cryptoAmountBaseUnit)
+          .div(bn(10).pow(underlyingAsset?.precision ?? asset.precision))
+          .toFixed(underlyingAsset?.precision ?? asset.precision),
```

Comment:
suggestion: `underlyingAsset.precision` is narrowed above, can be simplified (can always chain fp later on if needed)

```suggestion
toBaseUnit(opportunity.cryptoAmountBaseUnit, underlyingAsset.precision)
```

Reply:
Also not sure about this one here, looks like this should be `.times()` / `fromBaseUnit()` vs `.div()` / `toBaseUnit()`?

Reply:
Fixed in [4bade8f](https://github.com/shapeshift/web/pull/7644/commits/4bade8f9fe1ac3742ff2474ea6faa38eed2ad30e)

Used `fromBaseUnit`